pkg/sidecar/k8s_reactor.go

```diff
@@ -313,6 +315,25 @@ func (d *K8sReactor) manageContainer(ctx context.Context, container *docker.Cont
 			routeDst = r.Dst.String()
 		}
 
+		// detect the gateway => we know that we are going to delete the services CIDR, which are routed via the gateway,
+		// so we use they to figure out the gateway IP as well as the link index
+		if !addedGwRoute && routeDst == "100.64.0.0/10" {
```

Comment:
Let's move this out to a configuration variable and make it settable through a CLI flag?

Reply:
Totally agree about extracting it - done.

I don't think we need a CLI flag for this - this configuration is highly coupled with the exact configuration of the Kubernetes cluster, so chances are it won't work if you have a different configuration (this PR is a good example of how fragile networking updates are).